library-backend/library-app/src/test/java/org/internship/library/app/controller/BookControllerIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Integration Testing Class from Controller to DB
+ */
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {LibraryAppConfigTest.class})
+public class BookControllerIntegrationTest {
+
+    TestRestTemplate testRestTemplate = new TestRestTemplate();
+    HttpHeaders headers = new HttpHeaders();
+
+    /**
+     * Verify if get request return a book from database
+     */
+    @Test
+    public void getBook() {
+
+        String uri = "http://localhost:7777/library/book/{id}";
+        final String testBookId = "5";
+        ResponseEntity responseEntity = testRestTemplate.getForEntity(uri, BookEntity.class, testBookId);
+        assertNotNull(responseEntity);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+    }
+
+    /**
+     * Verify if post request persist a book in the database
+     */
+    @Test
+    void postBook() {
+
+        String uri = "http://localhost:7777/library/book";
+        final String testBookId = "5";
+        BookEntity testBook = new BookEntity();
+        testBook.setId(testBookId);
+        testBook.setTitle("Razv");
+        testBook.setAuthor("Cern");
+        testBook.setNumberOfPages(50);
+        ResponseEntity responseEntity = testRestTemplate.postForEntity(uri, testBook, BookEntity.class);
+
+        assertNotNull(responseEntity);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.CREATED);
+    }
+
+    /**
+     * Verify if update request update a book in the database
+     */
+    @Test
+    void updateBook() {
+
+        String uri = "http://localhost:7777/library/book/{id}";
+        final String testBookId = "5";
+        BookEntity testBook = new BookEntity();
+        testBook.setId(testBookId);
+        testBook.setTitle("Razv");
+        testBook.setAuthor("Cern");
+        testBook.setNumberOfPages(50);
+
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON));
+        HttpEntity<BookEntity> requestUpdate = new HttpEntity<>(testBook, headers);
+
+        ResponseEntity responseEntity = testRestTemplate.exchange(uri, HttpMethod.PUT, requestUpdate, BookEntity.class, testBookId);
+        assertNotNull(responseEntity);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
```

Comment:
Verify that the fields that you've wanted to update actually updated.

Flow:

get book 1
see the fields of book 1

update book 1 with some other author
verify update was done correctly

get book 1
verify the new author is the one provided by you at the update and not the initial one.